### ISSUE TITLE
race condition in TileDb.Run()

### ISSUE THREAD

**Issue by nkovacs** [fawick/go-mapnik#9]
m.qc is created inside a goroutine [here](https://github.com/fawick/go-mapnik/blob/master/maptiles/mbtilesgenerator.go#L114). It should be created before the goroutine is started.

**Comment by fawick** [comment]
Yes, you are right. It only can happen when Close() gets called prior to Run(), which is unlikely. Nevertheless, it indeed is a potential cause of error and I should fix that.
